### ISSUE TITLE
After a few days Pi Hole stops blocking ads

### ISSUE THREAD

**Issue by rj45jack** [pi-hole/pi-hole#409]
After a few days of leaving Pi Hole running this occurs: http://imgur.com/xY0fG9G

Then after running gravity.sh things start working again: http://imgur.com/yyWhbOL

Sometimes the "blocked domains" goes to "0". But the blocked ads always remains at "0" until I rerun the gravity script.

**Comment by kmr2319** [comment]
I've had similar problems where it would show 1 or 0 domains blocked.

**Comment by PromoFaux** [comment]
Hi there, can you try updating to the latest version (`curl -L https://install.pi-hole.net | bash`) to see if this problem is fixed. 

Many thanks

**Comment by Drjeffreyodweyer** [comment]
I got the exact same problem :( I updated two days ago (4/8/16) and today it happened again. Please help, I love my pi hole :)

**Comment by PromoFaux** [comment]
Hmmm. This is a strange one.

If you're all on the latest version, could you please run `pihole -d` and then paste the output (`/var/log/pihole_debug.log`) into a pastebin then link it here? thanks!

**Comment by rj45jack** [comment]
Just happened again: http://i.imgur.com/MpJ0gsB.png

` cat /var/log/pihole_debug.log
IP Addresses of this Pi:
192.168.0.18
127.0.0.1

Gateway check:
ok

#######################################
######## Whitelist Comparison #########
#######################################

#######################################
######## Blacklist Comparison #########
#######################################

#######################################
############ NSLookup Test ############
#######################################
NSLOOKUP of doubleclick.com from PiHole:
Server:         127.0.0.1
Address:        127.0.0.1#53

Name:   doubleclick.com
Address: 192.168.0.18


NSLOOKUP of doubleclick.com from 8.8.8.8:
Server:         8.8.8.8
Address:        8.8.8.8#53

Non-authoritative answer:
Name:   doubleclick.com
Address: 216.58.192.238


#######################################
########### Processes Check ###########
#######################################

lighttpd processes status:
● lighttpd.service - Lighttpd Daemon
   Loaded: loaded (/lib/systemd/system/lighttpd.service; enabled)
   Active: active (running) since Sun 2016-04-10 02:47:26 UTC; 9s ago
  Process: 7419 ExecStartPre=/usr/sbin/lighttpd -t -f /etc/lighttpd/lighttpd.conf (code=exited, status=0/SUCCESS)
 Main PID: 7427 (lighttpd)
   CGroup: /system.slice/lighttpd.service
           ├─7427 /usr/sbin/lighttpd -D -f /etc/lighttpd/lighttpd.conf
           ├─7441 /usr/bin/php-cgi
           ├─7442 /usr/bin/php-cgi
           ├─7443 /usr/bin/php-cgi
           ├─7444 /usr/bin/php-cgi
           └─7445 /usr/bin/php-cgi

Apr 10 02:47:26 kramer lighttpd[7419]: Syntax OK
Apr 10 02:47:26 kramer systemd[1]: Started Lighttpd Daemon.

dnsmasq processes status:
● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
   Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled)
  Drop-In: /run/systemd/generator/dnsmasq.service.d
           └─50-dnsmasq-$named.conf, 50-insserv.conf-$named.conf
   Active: active (running) since Sun 2016-04-10 02:47:25 UTC; 10s ago
  Process: 7254 ExecStop=/etc/init.d/dnsmasq systemd-stop-resolvconf (code=exited, status=0/SUCCESS)
  Process: 7327 ExecStartPost=/etc/init.d/dnsmasq systemd-start-resolvconf (code=exited, status=0/SUCCESS)
  Process: 7318 ExecStart=/etc/init.d/dnsmasq systemd-exec (code=exited, status=0/SUCCESS)
  Process: 7315 ExecStartPre=/usr/sbin/dnsmasq --test (code=exited, status=0/SUCCESS)
 Main PID: 7326 (dnsmasq)
   CGroup: /system.slice/dnsmasq.service
           └─7326 /usr/sbin/dnsmasq -x /var/run/dnsmasq/dnsmasq.pid -u dnsmasq -r /var/run/dnsmasq/resolv.conf -7 /etc/dnsmasq.d,.dpkg-dist,.dpkg-old,.dpkg-new --local-service --trust-anchor=.,19036,8,2,49AAC11D7B6F6446702E54A1607371607A1A41855200FD2CE1CDDE32F24E8FB5

Apr 10 02:47:25 kramer dnsmasq[7315]: dnsmasq: syntax check OK.
Apr 10 02:47:25 kramer systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.
#######################################
############### Dnsmasq ###############
#######################################

#######################################
########### 01-pihole.conf ############
#######################################
addn-hosts=/etc/pihole/gravity.list
domain-needed
bogus-priv
no-resolv
server=8.8.8.8
server=8.8.4.4
interface=eth0
listen-address=127.0.0.1
cache-size=10000
log-queries
log-facility=/var/log/pihole.log
local-ttl=300
log-async

#######################################
############ lighttpd.conf ############
#######################################
server.modules = (
"mod_access",
"mod_accesslog",
"mod_expire",
"mod_compress",
"mod_redirect",
"mod_setenv",
"mod_rewrite"
)
server.document-root        = "/var/www/html"
server.error-handler-404        = "pihole/index.html"
server.upload-dirs          = ( "/var/cache/lighttpd/uploads" )
server.errorlog             = "/var/log/lighttpd/error.log"
server.pid-file             = "/var/run/lighttpd.pid"
server.username             = "www-data"
server.groupname            = "www-data"
server.port                 = 80
accesslog.filename                      = "/var/log/lighttpd/access.log"
accesslog.format                        = "%{%s}t|%V|%r|%s|%b"
index-file.names            = ( "index.php", "index.html", "index.lighttpd.html" )
url.access-deny             = ( "~", ".inc" )
static-file.exclude-extensions = ( ".php", ".pl", ".fcgi" )
compress.cache-dir          = "/var/cache/lighttpd/compress/"
compress.filetype           = ( "application/javascript", "text/css", "text/html", "text/plain" )
include_shell "/usr/share/lighttpd/use-ipv6.pl " + server.port
include_shell "/usr/share/lighttpd/create-mime.assign.pl"
include_shell "/usr/share/lighttpd/include-conf-enabled.pl"
$HTTP["url"] =~ "^/admin/" {
setenv.add-response-header = ( "X-Pi-hole" => "The Pi-hole Web interface is working!" )
}
$HTTP["url"] =~ "^(?!/admin)/.*" {
setenv.add-response-header = ( "X-Pi-hole" => "A black hole for Internet advertisements." )
url.rewrite = ("(.*).js" => "pihole/index.js")
}

#######################################
############ gravity.list #############
#######################################
175050 /etc/pihole/gravity.list

Hostname of this pihole is:
kramer
#######################################
################ Hosts ################
#######################################
127.0.0.1       localhost
::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters

127.0.1.1       kramer

#######################################
############## Whitelist ##############
#######################################
adaway.org
adblock.gjtech.net
hosts-file.net
www.malwaredomainlist.com
pgl.yoyo.org
someonewhocares.org
winhelp2002.mvps.org
mirror1.malwaredomains.com
adblock.mahakala.is

#######################################
############## Blacklist ##############
#######################################

#######################################
############ adlists.list #############
#######################################
https://adaway.org/hosts.txt
http://adblock.gjtech.net/?format=unix-hosts
http://hosts-file.net/ad_servers.txt
http://www.malwaredomainlist.com/hostslist/hosts.txt
http://pgl.yoyo.org/adservers/serverlist.php?
http://someonewhocares.org/hosts/hosts
http://winhelp2002.mvps.org/hosts.txt
http://mirror1.malwaredomains.com/files/justdomains

#######################################
############# pihole.log ##############
#######################################
Apr 10 02:47:31 dnsmasq[7326]: query[A] d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com from 192.168.0.12
Apr 10 02:47:31 dnsmasq[7326]: cached d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com is NXDOMAIN
Apr 10 02:47:31 dnsmasq[7326]: query[A] d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com from 192.168.0.12
Apr 10 02:47:31 dnsmasq[7326]: cached d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com is NXDOMAIN
Apr 10 02:47:31 dnsmasq[7326]: query[A] d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com from 192.168.0.12
Apr 10 02:47:31 dnsmasq[7326]: cached d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com is NXDOMAIN
Apr 10 02:47:31 dnsmasq[7326]: query[A] d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com from 192.168.0.12
Apr 10 02:47:31 dnsmasq[7326]: cached d6-vodpf-01.mrdaily6.iptv.selfhost.corp.microsoft.com is NXDOMAIN
Apr 10 02:47:35 dnsmasq[7326]: query[A] doubleclick.com from 127.0.0.1
Apr 10 02:47:35 dnsmasq[7326]: /etc/pihole/gravity.list doubleclick.com is 192.168.0.18
`

**Comment by 0xc0ffea** [comment]
Just happened to me with
http://i.imgur.com/tfpIKjj.png

$ pihole -d
[sudo] password for trinity:
::: You are root... Beginning debug!
::: Writing local IPs to debug log
:::
::: Logging status of lighttpd and dnsmasq...
::: Writing dnsmasq.conf to debug log...
::: Writing 01-pihole.conf to debug log...
::: Writing lighttpd.conf to debug log...
::: Writing size of gravity.list to debug log...
::: Writing hosts file to debug log...
::: Writing whitelist to debug log...
::: Writing blacklist to debug log...
::: Writing adlists.list to debug log...
:::     No adlists.list file found... using adlists.default!
::: Writing current pihole traffic to debug log...
:::     Try loading any/all sites that you are having trouble with now...
:::     (Press ctrl+C to finish)



IP Addresses of this Pi:
10.0.0.2
127.0.0.1

Gateway check:
ok

#######################################
######## Whitelist Comparison #########
#######################################

#######################################
######## Blacklist Comparison #########
#######################################

#######################################
############ NSLookup Test ############
#######################################
NSLOOKUP of doubleclick.com from PiHole:
Server:         127.0.0.1
Address:        127.0.0.1#53

Non-authoritative answer:
Name:   doubleclick.com
Address: 172.217.3.14


NSLOOKUP of doubleclick.com from 8.8.8.8:
Server:         8.8.8.8
Address:        8.8.8.8#53

Non-authoritative answer:
Name:   doubleclick.com
Address: 172.217.3.14


#######################################
########### Processes Check ###########
#######################################

lighttpd processes status:
● lighttpd.service - Lighttpd Daemon
   Loaded: loaded (/lib/systemd/system/lighttpd.service; enabled)
   Active: active (running) since Fri 2016-04-08 17:17:15 EDT; 1 day 17h ago
  Process: 601 ExecStartPre=/usr/sbin/lighttpd -t -f /etc/lighttpd/lighttpd.conf (code=exited, status=0/SUCCESS)
 Main PID: 828 (lighttpd)
   CGroup: /system.slice/lighttpd.service
           ├─828 /usr/sbin/lighttpd -D -f /etc/lighttpd/lighttpd.conf
           ├─873 /usr/bin/php-cgi
           ├─904 /usr/bin/php-cgi
           ├─906 /usr/bin/php-cgi
           ├─907 /usr/bin/php-cgi
           └─908 /usr/bin/php-cgi

Apr 08 17:17:15 Cerberus lighttpd[601]: Syntax OK
Apr 08 17:17:15 Cerberus systemd[1]: Started Lighttpd Daemon.

dnsmasq processes status:
● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
   Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled)
  Drop-In: /run/systemd/generator/dnsmasq.service.d
           └─50-dnsmasq-$named.conf, 50-insserv.conf-$named.conf
   Active: active (running) since Fri 2016-04-08 17:17:14 EDT; 1 day 17h ago
  Process: 709 ExecStartPost=/etc/init.d/dnsmasq systemd-start-resolvconf (code=exited, status=0/SUCCESS)
  Process: 585 ExecStart=/etc/init.d/dnsmasq systemd-exec (code=exited, status=0/SUCCESS)
  Process: 537 ExecStartPre=/usr/sbin/dnsmasq --test (code=exited, status=0/SUCCESS)
 Main PID: 600 (dnsmasq)
   CGroup: /system.slice/dnsmasq.service
           └─600 /usr/sbin/dnsmasq -x /var/run/dnsmasq/dnsmasq.pid -u dnsmasq -r /var/run/dnsmasq/resolv.conf -7 /etc/dnsmasq.d,.dpkg-dist,.dpkg-old,.dpkg-new --local-service --trust-anchor=.,19036,8,2,49AAC11D7B6F6446702E54A1607371607A1A41855200FD2CE1CDDE32F24E8FB5

Apr 08 17:17:13 Cerberus dnsmasq[537]: dnsmasq: syntax check OK.
Apr 08 17:17:14 Cerberus systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.
#######################################
############### Dnsmasq ###############
#######################################

#######################################
########### 01-pihole.conf ############
#######################################
addn-hosts=/etc/pihole/gravity.list
domain-needed
bogus-priv
no-resolv
server=8.8.8.8
server=8.8.4.4
interface=eth0
listen-address=127.0.0.1
cache-size=10000
log-queries
log-facility=/var/log/pihole.log
local-ttl=300
log-async

#######################################
############ lighttpd.conf ############
#######################################
server.modules = (
"mod_access",
"mod_accesslog",
"mod_expire",
"mod_compress",
"mod_redirect",
"mod_setenv",
"mod_rewrite"
)
server.document-root        = "/var/www/html"
server.error-handler-404        = "pihole/index.html"
server.upload-dirs          = ( "/var/cache/lighttpd/uploads" )
server.errorlog             = "/var/log/lighttpd/error.log"
server.pid-file             = "/var/run/lighttpd.pid"
server.username             = "www-data"
server.groupname            = "www-data"
server.port                 = 80
accesslog.filename                      = "/var/log/lighttpd/access.log"
accesslog.format                        = "%{%s}t|%V|%r|%s|%b"
index-file.names            = ( "index.php", "index.html", "index.lighttpd.html" )
url.access-deny             = ( "~", ".inc" )
static-file.exclude-extensions = ( ".php", ".pl", ".fcgi" )
compress.cache-dir          = "/var/cache/lighttpd/compress/"
compress.filetype           = ( "application/javascript", "text/css", "text/html", "text/plain" )
include_shell "/usr/share/lighttpd/use-ipv6.pl " + server.port
include_shell "/usr/share/lighttpd/create-mime.assign.pl"
include_shell "/usr/share/lighttpd/include-conf-enabled.pl"
$HTTP["url"] =~ "^/admin/" {
setenv.add-response-header = ( "X-Pi-hole" => "The Pi-hole Web interface is working!" )
}
$HTTP["url"] =~ "^(?!/admin)/.*" {
setenv.add-response-header = ( "X-Pi-hole" => "A black hole for Internet advertisements." )
url.rewrite = ("(.*).js" => "pihole/index.js")
}

#######################################
############ gravity.list #############
#######################################
4 /etc/pihole/gravity.list

Hostname of this pihole is:
Cerberus
#######################################
################ Hosts ################
#######################################
127.0.0.1       localhost
::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters

127.0.1.1       cerberus

#######################################
############## Whitelist ##############
#######################################
raw.githubusercontent.com
adblock.gjtech.net
mirror1.malwaredomains.com
sysctl.org
zeustracker.abuse.ch
s3.amazonaws.com
hosts-file.net


ui.skype.com
apps.skype.com
m.hotmail.com
s.gateway.messenger.live.com

#######################################
############## Blacklist ##############
#######################################

#######################################
############ adlists.list #############
#######################################
No adlists.list file found... using adlists.default!
#######################################
############# pihole.log ##############
#######################################
Apr 10 10:21:57 dnsmasq[600]: reply safebrowsing-cache.google.com is <CNAME>
Apr 10 10:21:57 dnsmasq[600]: reply safebrowsing.cache.l.google.com is 172.217.4.46
Apr 10 10:21:57 dnsmasq[600]: reply safebrowsing-cache.google.com is <CNAME>
Apr 10 10:21:57 dnsmasq[600]: reply safebrowsing.cache.l.google.com is 2607:f8b0:4006:80e::200e
Apr 10 10:22:08 dnsmasq[600]: query[PTR] 15.0.0.10.in-addr.arpa from 127.0.0.1
Apr 10 10:22:08 dnsmasq[600]: config 10.0.0.15 is NXDOMAIN
Apr 10 10:22:17 dnsmasq[600]: query[A] doubleclick.com from 127.0.0.1
Apr 10 10:22:17 dnsmasq[600]: forwarded doubleclick.com to 8.8.4.4
Apr 10 10:22:17 dnsmasq[600]: forwarded doubleclick.com to 8.8.8.8
Apr 10 10:22:17 dnsmasq[600]: reply doubleclick.com is 172.217.3.14
Apr 10 10:22:29 dnsmasq[600]: query[AAAA] eu.patch.battle.net from 2601:49:4101:cee8:60ea:2f80:6c18:c9fa
Apr 10 10:22:29 dnsmasq[600]: forwarded eu.patch.battle.net to 8.8.4.4
Apr 10 10:22:29 dnsmasq[600]: reply eu.patch.battle.net is NODATA-IPv6
Apr 10 10:22:32 dnsmasq[600]: query[A] clients4.google.com from 2601:49:4101:cee8:d94b:1e8e:77e:9890
Apr 10 10:22:32 dnsmasq[600]: forwarded clients4.google.com to 8.8.4.4
Apr 10 10:22:32 dnsmasq[600]: query[AAAA] clients4.google.com from 2601:49:4101:cee8:d94b:1e8e:77e:9890
Apr 10 10:22:32 dnsmasq[600]: forwarded clients4.google.com to 8.8.4.4
Apr 10 10:22:32 dnsmasq[600]: reply clients4.google.com is <CNAME>
Apr 10 10:22:32 dnsmasq[600]: reply clients.l.google.com is 172.217.4.78
Apr 10 10:22:32 dnsmasq[600]: reply clients4.google.com is <CNAME>
Apr 10 10:22:32 dnsmasq[600]: reply clients.l.google.com is 2607:f8b0:4006:808::200e
Apr 10 10:22:36 dnsmasq[600]: query[A] www.paypal.com from 2601:49:4101:cee8:d94b:1e8e:77e:9890
Apr 10 10:22:36 dnsmasq[600]: cached www.paypal.com is <CNAME>
Apr 10 10:22:36 dnsmasq[600]: forwarded www.paypal.com to 8.8.4.4
Apr 10 10:22:36 dnsmasq[600]: query[AAAA] www.paypal.com from 2601:49:4101:cee8:d94b:1e8e:77e:9890
Apr 10 10:22:36 dnsmasq[600]: forwarded www.paypal.com to 8.8.4.4
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply ppdirect.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply wlb.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com.edgekey.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply e3694.a.akamaiedge.net is NODATA-IPv6
Apr 10 10:22:37 dnsmasq[600]: query[A] www.paypal.com from 10.0.0.15
Apr 10 10:22:37 dnsmasq[600]: cached www.paypal.com is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: cached www.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: cached ppdirect.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: cached wlb.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: cached www.paypal.com.edgekey.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: forwarded www.paypal.com to 8.8.4.4
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply ppdirect.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply wlb.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com.edgekey.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply e3694.a.akamaiedge.net is 104.85.153.224
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply ppdirect.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply wlb.paypal.com.akadns.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply www.paypal.com.edgekey.net is <CNAME>
Apr 10 10:22:37 dnsmasq[600]: reply e3694.a.akamaiedge.net is 104.85.153.224
Apr 10 10:23:04 dnsmasq[600]: query[A] trevor-producer-cdn.api.bbci.co.uk from 10.0.0.8
Apr 10 10:23:04 dnsmasq[600]: forwarded trevor-producer-cdn.api.bbci.co.uk to 8.8.4.4
Apr 10 10:23:04 dnsmasq[600]: forwarded trevor-producer-cdn.api.bbci.co.uk to 8.8.8.8
Apr 10 10:23:04 dnsmasq[600]: reply trevor-producer-cdn.api.bbci.co.uk is <CNAME>
Apr 10 10:23:04 dnsmasq[600]: reply newsapps-live.bbc.co.uk.edgekey.net is <CNAME>
Apr 10 10:23:04 dnsmasq[600]: reply e19.e12.akamaiedge.net is 23.196.112.45
Apr 10 10:23:07 dnsmasq[600]: query[A] clients4.google.com from 2601:49:4101:cee8:60ea:2f80:6c18:c9fa
Apr 10 10:23:07 dnsmasq[600]: cached clients4.google.com is <CNAME>
Apr 10 10:23:07 dnsmasq[600]: cached clients.l.google.com is 172.217.4.78
Apr 10 10:23:07 dnsmasq[600]: query[AAAA] clients4.google.com from 2601:49:4101:cee8:60ea:2f80:6c18:c9fa
Apr 10 10:23:07 dnsmasq[600]: cached clients4.google.com is <CNAME>
Apr 10 10:23:07 dnsmasq[600]: cached clients.l.google.com is 2607:f8b0:4006:808::200e
Apr 10 10:23:16 dnsmasq[600]: query[A] clientconfig.akamai.steamstatic.com from 2601:49:4101:cee8:60ea:2f80:6c18:c9fa
Apr 10 10:23:16 dnsmasq[600]: cached clientconfig.akamai.steamstatic.com is <CNAME>
Apr 10 10:23:16 dnsmasq[600]: cached akacdn.valve.com.edgesuite.net is <CNAME>
Apr 10 10:23:16 dnsmasq[600]: forwarded clientconfig.akamai.steamstatic.com to 8.8.4.4
Apr 10 10:23:16 dnsmasq[600]: query[A] clientconfig.akamai.steamstatic.com from 10.0.0.13
Apr 10 10:23:16 dnsmasq[600]: cached clientconfig.akamai.steamstatic.com is <CNAME>
Apr 10 10:23:16 dnsmasq[600]: forwarded clientconfig.akamai.steamstatic.com to 8.8.4.4
Apr 10 10:23:16 dnsmasq[600]: reply clientconfig.akamai.steamstatic.com is <CNAME>
Apr 10 10:23:16 dnsmasq[600]: reply akacdn.valve.com.edgesuite.net is <CNAME>
Apr 10 10:23:16 dnsmasq[600]: reply a1507.d.akamai.net is 23.217.129.64
Apr 10 10:23:16 dnsmasq[600]: reply a1507.d.akamai.net is 23.217.129.48
Apr 10 10:23:16 dnsmasq[600]: reply clientconfig.akamai.steamstatic.com is <CNAME>
Apr 10 10:23:16 dnsmasq[600]: reply akacdn.valve.com.edgesuite.net is <CNAME>
Apr 10 10:23:16 dnsmasq[600]: reply a1507.d.akamai.net is 23.217.129.64
Apr 10 10:23:16 dnsmasq[600]: reply a1507.d.akamai.net is 23.217.129.32
Apr 10 10:24:14 dnsmasq[600]: query[AAAA] www.google-analytics.com from 2601:49:4101:cee8:d94b:1e8e:77e:9890
Apr 10 10:24:14 dnsmasq[600]: cached www.google-analytics.com is <CNAME>
Apr 10 10:24:14 dnsmasq[600]: forwarded www.google-analytics.com to 8.8.4.4
Apr 10 10:24:14 dnsmasq[600]: forwarded www.google-analytics.com to 8.8.8.8
Apr 10 10:24:14 dnsmasq[600]: reply www.google-analytics.com is <CNAME>
Apr 10 10:24:14 dnsmasq[600]: reply www-google-analytics.l.google.com is 2607:f8b0:4006:806::200e
Apr 10 10:24:14 dnsmasq[600]: query[A] dynupdate.no-ip.com from 127.0.0.1
Apr 10 10:24:14 dnsmasq[600]: cached dynupdate.no-ip.com is 8.23.224.120
Apr 10 10:24:18 dnsmasq[600]: query[A] github.com from 2601:49:4101:cee8:d94b:1e8e:77e:9890
Apr 10 10:24:18 dnsmasq[600]: forwarded github.com to 8.8.4.4
Apr 10 10:24:18 dnsmasq[600]: reply github.com is 192.30.252.121
Apr 10 10:24:31 dnsmasq[600]: query[A] api.github.com from 2601:49:4101:cee8:d94b:1e8e:77e:9890
Apr 10 10:24:31 dnsmasq[600]: forwarded api.github.com to 8.8.4.4
Apr 10 10:24:31 dnsmasq[600]: reply api.github.com is 192.30.252.124


Fixed by reinstalled :/

**Comment by Mcat12** [comment]
@rj45jack and @0xc0ffea, can you please copy the log instead to http://pastebin.com and link it? It helps clean up the conversation.

**Comment by articuno1au** [comment]
I've just come across this issue also. I noticed I was being served ads. When I loaded the admin panel, I was shown that there were 3 domains being blacklisted.

I fixed by rerunning the gravity script (pihole -g).

I'm running my Pi-Hole on Debian hosted in a VM, so not exactly a normal setup. I updated from 2.5.3 to 2.6.2, and only encountered the issue a few days after the upgrade process.

I had fixed the problem prior to seeing this issue thread, so unfortunately I don't have a dump file for you. If it recurs I will get you one.

**Comment by Mcat12** [comment]
O.o It's happening here too, I think... 
![Screenshot](http://i.imgur.com/OUFGyVM.png)

**Comment by Mcat12** [comment]
It appears that only the hostname of the pi, `pi.hole` (another way to reach the pi... `pi.hole/admin`), and the blacklist are present in the blocklist. So the actual lists and the whitelist do not get added...

**Comment by Mcat12** [comment]
After running gravity, it works... Weird.

**Comment by PromoFaux** [comment]
OK, so it sounds like this may be down to permissions. I thought the cron job ran as root, though?

**Comment by SteveW25561** [comment]
Same issue here: suddenly down to 3 domains being blocked.  I re-ran the full installer before seeing this thread.  I was (and am a still) on v2.6.1.2

**Comment by TechnicallyFox** [comment]
I just started getting this problem after upgrading from 2.5.3 and confirmed it on two other devices. Runs fine for couple days then down to 3 or 4 blocked domains until gravity is run.

**Comment by PromoFaux** [comment]
Right, I think we can definitely put this down to the `cron` job, as it seems to have happened at the same(ish) time for everyone. (The [`cron` job](https://github.com/pi-hole/pi-hole/blob/master/advanced/pihole.cron#L12-L14) is set to run at 01:59 every Sunday)

Interestingly, this hasn't happened on my setup, but I'm not using a raspberry pi............

**Comment by DonLexos** [comment]
Had the same issue, quick fis ix running `pihole -g` (gravity list update)

**Comment by PromoFaux** [comment]
Permissions after running `pihole -g`
![image](https://cloud.githubusercontent.com/assets/1998970/14421606/78951a3a-ffca-11e5-8caa-40d4585b1cad.png)

Permissions after cron runs the same command:
![image](https://cloud.githubusercontent.com/assets/1998970/14421958/2bf29a34-ffcc-11e5-83d5-b258bc325fc2.png)

Now to find the bit that is removing write permissions from `gravity.list`... I think that may be the issue here!

**Comment by articuno1au** [comment]
Out of curiosity, how did you run the cron script? 

I haven't been able to replicate. After running the cron script I'm still seeing:
![image](https://cloud.githubusercontent.com/assets/18381114/14422222/df4f43ce-0018-11e6-9073-5895bf415ff1.png)

**Comment by PromoFaux** [comment]
I cheated, I set it to run every minute and rebooted the machine!

![image](https://cloud.githubusercontent.com/assets/1998970/14422281/ed2eb952-ffcd-11e5-9cdc-cb0e6740499d.png)

**Comment by articuno1au** [comment]
Hax. Will follow you down the rabbit hole to see if I can replicate.

Well that's interesting. Note the file size of the gravity list and the permissions:
![image](https://cloud.githubusercontent.com/assets/18381114/14422416/122a731c-001a-11e6-8537-fda2dc6ab181.png)
Then 30 seconds later (before the cron script re-executes)
![image](https://cloud.githubusercontent.com/assets/18381114/14422464/46688042-001a-11e6-9372-4e80e33b98a5.png)

It's like it's rebuilding the file in the background before cron fires again (which I know is because I just set it to run each minute).

**Comment by PromoFaux** [comment]
So the way gravity works is like so:

- Download/update source lists from various locations
- Parse them all into a single file
- Process that file several times to remove comments/sort the domains out etc
- Copy the final file (`pihole.3.accretionDisc.txt`) over the top of `gravity.list` _(I think this might be where it's going wrong. I might be wrong though)_
- Clear up un-used `pihole.n.*` files

It may be that we just need to preface [this line](https://github.com/pi-hole/pi-hole/blob/3f0b15902d89ae3d062499fa289d15195069bf9b/gravity.sh#L282) with a `$SUDO`, however as `cron` runs the script as `root`, that shouldn't matter

**Comment by articuno1au** [comment]
Trying that SS again:
![image](https://cloud.githubusercontent.com/assets/18381114/14422705/75949670-001b-11e6-84ef-c05bc4de2d4d.png)

**Comment by PromoFaux** [comment]
I had not thought to tail `gravity.list`! Yeah, as per my previous post, It get's overwritten when gravity runs.

**Comment by articuno1au** [comment]
I'm getting realllly weird outputs from tails now.

I've just fixed everything on my end. If you need more testing or anything, please give me a shout. In the meantime, food :)

**Comment by PromoFaux** [comment]
It's strange, according to `mail`, the cron job is getting different numbers of domains on each run of `pihole -g`

I have some stuff at my actual job to be getting on with, but I will give it some thought and let you know if I need anything! Thanks!

**Comment by PromoFaux** [comment]
**Update**

Found it! Its the `spinner` function that is causing the issues, weirdly. It's kind of hacked in anyway, so I'll remove it.

**Comment by PromoFaux** [comment]
@articuno1au Can I borrow you for testing purposes, please?

```
cd /etc/.pihole
sudo git fetch
sudo git checkout BugFix409
./automated\ install/basic-install.sh
```

Then make the changes to the cron job again as we did earlier and confirm that it's fixed! 
Thanks

**Comment by articuno1au** [comment]
Testing now. Will edit with results.

**Comment by articuno1au** [comment]
Looking good.
![image](https://cloud.githubusercontent.com/assets/18381114/14425465/d2a611a4-002a-11e6-9270-619c52af6579.png)
![image](https://cloud.githubusercontent.com/assets/18381114/14425482/e5153d2e-002a-11e6-8cd9-55349dd4f174.png)
I'd be calling it fixed.

Nice work sir :)

**Comment by PromoFaux** [comment]
Super, that's good enough for me. Will merge and release! Hold your horses!

(by the way, just `sudo git checkout master` and re-run the install script again to go back to main branch.)

**Comment by articuno1au** [comment]
Cheers.

I had already done so to save me having to nano into my cron files again. Advanced laziness ;)

**Comment by PromoFaux** [comment]
OK! Fixed in 2.6.2! Re-run install script to update (next up.. a working update script. #soon)

**Comment by articuno1au** [comment]
@PromoFaux - Apologies if this re-opens (I'm a bitbucket user).

If you need help testing anything in the future, just mention me and I'll come check it out. I'm in GMT+10, so I should be off work when you're pushing stuff.
